website/project/model.py

```diff
@@ -2829,188 +2894,302 @@ def to_json(self):
             "anonymous": self.anonymous
         }
 
+class Sanction(StoredObject):
+    """Sanction object is a generic way to track approval states"""
 
-def validate_retraction_state(value):
-    acceptable_states = [Retraction.PENDING, Retraction.RETRACTED, Retraction.CANCELLED]
-    if value not in acceptable_states:
-        raise ValidationValueError('Invalid retraction state assignment.')
+    abstract = True
 
-    return True
+    UNAPPROVED = 'unapproved'
+    APPROVED = 'approved'
+    REJECTED = 'rejected'
 
+    DISPLAY_NAME = 'sanction'
 
-class Retraction(StoredObject):
-    """Retraction object for public registrations."""
-
-    PENDING = 'pending'
-    RETRACTED = 'retracted'
-    CANCELLED = 'cancelled'
+    APPROVAL_NOT_AUTHORIZED_MESSAGE = 'This user is not authorized to approve this {0}'
```

Comment:
Minor, but using keyword rather than positional string formatting would be more self-documenting:

```python
 'This user is not authorized to approve this {display_name}'
```